internal/edgedbtypes/datetime_test.go

```diff
@@ -69,3 +69,71 @@ func TestNewLocalDateTime(t *testing.T) {
 		})
 	}
 }
+
+func TestDurationString(t *testing.T) {
+	samples := []struct {
+		str string
+		d   Duration
+	}{
+		{"1μs", Duration(1)},
+		{"999μs", Duration(999)},
+		{"-1μs", Duration(-1)},
+		{"-999μs", Duration(-999)},
+
+		{"1.234ms", Duration(1_234)},
+		{"1.2ms", Duration(1_200)},
+		{"1.004ms", Duration(1_004)},
+		{"999.234ms", Duration(999_234)},
+		{"999.2ms", Duration(999_200)},
+		{"999.004ms", Duration(999_004)},
+		{"-1.234ms", Duration(-1_234)},
+		{"-1.2ms", Duration(-1_200)},
+		{"-1.004ms", Duration(-1_004)},
+		{"-999.234ms", Duration(-999_234)},
+		{"-999.2ms", Duration(-999_200)},
+		{"-999.004ms", Duration(-999_004)},
+
+		{"0s", Duration(0)},
+		{"1s", Duration(1_000_000)},
+		{"59s", Duration(59_000_000)},
+		{"1.234567s", Duration(1_234_567)},
+		{"1.2s", Duration(1_200_000)},
+		{"1.000007s", Duration(1_000_007)},
+		{"59.234567s", Duration(59_234_567)},
+		{"59.2s", Duration(59_200_000)},
+		{"59.000007s", Duration(59_000_007)},
+		{"-1s", Duration(-1_000_000)},
+		{"-59s", Duration(-59_000_000)},
+		{"-1.234567s", Duration(-1_234_567)},
+		{"-1.2s", Duration(-1_200_000)},
+		{"-1.000007s", Duration(-1_000_007)},
+		{"-59.234567s", Duration(-59_234_567)},
+		{"-59.2s", Duration(-59_200_000)},
+		{"-59.000007s", Duration(-59_000_007)},
+
+		{"1m", Duration(60_000_000)},
+		{"59m", Duration(3540000000)},
+		{"-1m", Duration(-60_000_000)},
+		{"-59m", Duration(-3540000000)},
+
+		{"1h", Duration(3600000000)},
+		{"24h", Duration(86400000000)},
+		{"-1h", Duration(-3600000000)},
+		{"-24h", Duration(-86400000000)},
+
+		{"59m59s", Duration(3599000000)},
+		{"1h59m59s", Duration(7199000000)},
+		{"1h59m", Duration(7140000000)},
+		{"854015929h20m18.258432s", Duration(3074457345618258432)},
+		{"-59m59s", Duration(-3599000000)},
+		{"-1h59m59s", Duration(-7199000000)},
+		{"-1h59m", Duration(-7140000000)},
+		{"-854015929h20m18.258432s", Duration(-3074457345618258432)},
+	}
+
+	for _, s := range samples {
+		t.Run(s.str, func(t *testing.T) {
+			assert.Equal(t, s.str, s.d.String())
```

Comment:
Can you also add a functional test with EdgeDB, where you send duration converted from a string like "-1h35m17s" and convert it to string in EdgeQL, and then compare the result on the client side?

Reply:
added :heavy_check_mark: 

I  had to change `μs` to `us` because of https://github.com/edgedb/edgedb/issues/2194 and I removed a test case because of https://github.com/edgedb/edgedb/issues/2195